### ISSUE TITLE
New request: Camera still image (http)

### ISSUE THREAD

**Issue by aferraz** [Filipowicz251/mijia-1080P-hacks#12]
Hi,
 it's possible to add a web server returning a still image from the camera ?
Something like http://IP_OF_CAMERA/image.jpg

I would like to use this camera in homeassistant automation software. With a still image I can make an automation where I could send the image via Telegram, for example, in case motion is triggered.

Thanks
 Alessandro

**Comment by willthrom** [comment]
I don't see any reason you couldn't implement that. Usually a short video it is more useful for motion detection thought

**Comment by aferraz** [comment]
You are right mut at the moment homeassistant don't support videos on Telegram notifications.

**Comment by itanczos** [comment]
@aferraz 
HomeAssistant handles RTSP stream via FFMPEG.

**Comment by aferraz** [comment]
@itanczos 
I know but I don't find a way to send it via push notification.

**Comment by bartvollebregt** [comment]
I would love to see this implemented. I would then be able to easily integrate it in my OpenHab2 installations without having to do rtsp > jpeg conversions with ffmpeg....

**Comment by willthrom** [comment]
Having in the camera a software running a webservice publishing Image on Demand is possible but it involves quite a lot of work and I don't think someone will implement it..

The real problem is what imagine you want to publish? Whatever is there? What value has that?

I could understand a motion event triggered and the camera send that notification (Image, like upload the video/image to FTP server or send an email).
Again, a lot of work. Currently I am personally focusing in an proper RTSP Server.....

@aferraz Change the issue name to something more meaningful please.

**Comment by bartvollebregt** [comment]
@willthrom I think something like this was implemented in one of the other xiaomi camera hacks. I'll try to find it. 

They way it works is that its just a web server running on an url like http://yourip/image_latest.jpg

Everytime some does a request it will capture an image and put in on a webserver on the camera so image_latest.jpg is always the latest image.

**Comment by kollaesch** [comment]
@bartvollebregt 
How about this: I'd prefer to this keep this slim. So having a `custom_component`|`script`in hass would be an alternative.

check [here](https://community.home-assistant.io/t/save-camera-image-when-motion-detected/12556/8)

**Comment by aferraz** [comment]
I use a script in HA and it work.
In my case the camera is the problem.
Sometimes after a reboot it remains with the orange led on and is unreachable.

**Comment by AndBaetz** [comment]
Would be really cool for openHAB2 to have that implemented.

**Comment by bgeneto** [comment]
I would love this feature as well because Domoticz only have support for http (or https) still images. In other hacks, like fang-hacks, this feature is relative easy to implement but with this present hack for mijia  I'm completely lost... Any tips? I see that there is an option to turn on http server, just don't know how to generate jpg still images every second.
